lib/helpers/query/cast$expr.js

```diff
@@ -0,0 +1,229 @@
+    val.$cond.else = _castExpression(val.$cond.else, schema, strictQuery);
+  } else if (val.$ifNull != null) {
+    val.$ifNull.map(v => _castExpression(v, schema, strictQuery));
+  } else if (val.$switch != null) {
+    val.branches.map(v => _castExpression(v, schema, strictQuery));
+    val.default = _castExpression(val.default, schema, strictQuery);
+  }
+
+  const keys = Object.keys(val);
+  for (const key of keys) {
+    if (booleanComparison.has(key)) {
+      val[key] = val[key].map(v => _castExpression(v, schema, strictQuery));
+    } else if (comparisonOperator.has(key)) {
+      val[key] = castComparison(val[key], schema, strictQuery);
+    } else if (arithmeticOperatorArray.has(key)) {
+      val[key] = castArithmetic(val[key], schema, strictQuery);
+    } else if (arithmeticOperatorNumber.has(key)) {
+      val[key] = castArithmeticSingle(val[key], schema, strictQuery);
+    }
+  }
+
+  _omitUndefined(val);
+
+  return val;
+}
+
+function _omitUndefined(val) {
+  const keys = Object.keys(val);
+  for (const key of keys) {
+    if (val[key] === void 0) {
+      delete val[key];
+    }
+  }
+}
+
+// { $op: <number> }
+function castArithmeticSingle(val) {
+  if (!isLiteral(val)) {
+    return val;
+  }
+
+  try {
+    return castNumber(val);
+  } catch (err) {
+    throw new CastError('Number', val);
+  }
+}
+
+// { $op: [<number>, <number>] }
+function castArithmetic(val) {
+  if (!Array.isArray(val)) {
+    if (!isLiteral(val)) {
+      return val;
+    }
+    try {
+      return castNumber(val);
+    } catch (err) {
+      throw new CastError('Number', val);
+    }
+  }
+
+  return val.map(v => {
+    if (!isLiteral(v)) {
+      return v;
+    }
+    try {
+      return castNumber(v);
+    } catch (err) {
+      throw new CastError('Number', v);
+    }
+  });
+}
+
+// { $op: [expression, expression] }
+function castComparison(val, schema, strictQuery) {
+  if (!Array.isArray(val) || val.length !== 2) {
+    throw new Error('Comparison operator must be an array of length 2');
+  }
+
+  val[0] = _castExpression(val[0], schema, strictQuery);
+  const lhs = val[0];
+
+  if (isLiteral(val[1])) {
+    let path = null;
+    let schematype = null;
+    let caster = null;
+    if (isPath(lhs)) {
+      path = lhs.slice(1);
+      schematype = schema.path(path);
+    } else if (typeof lhs === 'object' && lhs != null) {
+      if (isPath(lhs.$year)) {
+        path = lhs.$year.slice(1) + '.$year';
+        caster = castNumber;
+      } else if (isPath(lhs.$month)) {
+        path = lhs.$month.slice(1) + '.$month';
+        caster = castNumber;
+      } else if (isPath(lhs.$week)) {
+        path = lhs.$week.slice(1) + '.$week';
+        caster = castNumber;
+      } else if (isPath(lhs.$dayOfMonth)) {
+        path = lhs.$dayOfMonth.slice(1) + '.$dayOfMonth';
+        caster = castNumber;
+      } else if (isPath(lhs.$dayOfYear)) {
+        path = lhs.$dayOfMonth.slice(1) + '.$dayOfYear';
```

Comment:
```suggestion
        path = lhs.$dayOfYear.slice(1) + '.$dayOfYear';
```

Reply:
Fixed in 29b323c by removing this repetitive `if`